ocean-client/src/programs/vault-maxi-program.ts

```diff
@@ -43,26 +57,46 @@ export class VaultMaxiProgram extends CommonProgram {
         }
         const stock_per_token = +pool!.tokenA.reserve / +pool!.totalLiquidity.token
         const removeTokens = Math.min(neededStock / stock_per_token, lptokens)
+        await this.updateToState(ProgramState.DoingTransaction, VaultMaxiProgramTransaction.RemoveLiquidity)
         console.log(" would need " + (neededStock / stock_per_token) + " doing " + removeTokens + " ")
         const removeTx = await this.removeLiquidity(+pool!.id, new BigNumber(removeTokens))
+
+        await this.updateToState(ProgramState.WaitingForLastTransaction, VaultMaxiProgramTransaction.RemoveLiquidity)
         if (! await this.waitForTx(removeTx)) {
             await telegram.send("ERROR: when removing liquidity")
             return false
         }
         const tokens = await this.getTokenBalances()
         console.log(" removed liq. got tokens: " + Array.from(tokens.values()).map(value => " " + value.amount + "@" + value.symbol))
-        let paybackTokens: TokenBalance[] = []
+        
+        let paybackTokens: AddressToken[] = []
         let token = tokens.get("DUSD")
-        if (token) paybackTokens.push({ token: +token.id, amount: new BigNumber(Math.min(+token.amount, wantedusd)) })
+        if (token) {
+            token.amount = "" + Math.min(+token.amount, wantedusd)
+            paybackTokens.push(token)
+        }
         token = tokens.get(this.settings.LMToken)
-        if (token) paybackTokens.push({ token: +token.id, amount: new BigNumber(Math.min(+token.amount, neededStock)) })
+        if (token) {
+            token.amount = "" + Math.min(+token.amount, neededStock)
+            paybackTokens.push(token)
+        }
 
+        return await this.paybackTokenBalances(paybackTokens, telegram)
+    }
 
-        console.log(" paying back tokens " + paybackTokens.map(token => " " + token.amount + "@" + token.token))
+    private async paybackTokenBalances(addressTokens: AddressToken[], telegram: Telegram): Promise<boolean> {
+        let paybackTokens: TokenBalance[] = []
+        addressTokens.forEach(addressToken => {
+            paybackTokens.push({ token: +addressToken.id, amount: new BigNumber(addressToken.amount) })
+        })
+        console.log(" paying back tokens " + addressTokens.map(token => " " + token.amount + "@" + token.symbol))
         if (paybackTokens.length > 0) {
+            await this.updateToState(ProgramState.DoingTransaction, VaultMaxiProgramTransaction.PaybackLoan)
             const paybackTx = await this.paybackLoans(paybackTokens)
-            const sucess = await this.waitForTx(paybackTx)
-            if (!sucess) {
+
+            await this.updateToState(ProgramState.WaitingForLastTransaction, VaultMaxiProgramTransaction.PaybackLoan)
```

Comment:
@kuegi I haven't missed paybackloan they are in that function, which is reused. We can debate if this is correct, as clean up could fail too